File: py/solns/binaryTreePostorderTraversal/binaryTreePostorderTraversal.py
from solns.treeNode.treeNode import *
class Solution:
    @staticmethod
    def naive(root):
        res = []
        def postOrder(tree):
            if tree==None: return 
            postOrder(tree.left)
            postOrder(tree.right)
            res.append(tree.val) 
        postOrder(root)
        return res

File: cpp/example/src/BinaryTreePostorderTraversal/binaryTreePostorderTraversal.cpp
#include "binaryTreePostorderTraversal.h"

void binaryTreePostorderTraversal::recursive(TreeNode* tree){
    if (tree==nullptr) return;
    recursive(tree->left);
    recursive(tree->right);
    this->res.push_back(tree->val);
}

vector<int> binaryTreePostorderTraversal::naive(TreeNode* root){
    recursive(root);
    return this->res;
}
